File: solutions/javascript/bob/1/bob.js
//
// This is only a SKELETON file for the 'Bob' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const hey = (message) => {
  // Handle empty or null messages
  if (message === null || message.trim() === "") {
    return 'Fine. Be that way!';
  }

  // Check if the message is in all uppercase (shouting) and contains at least one letter
  if (message === message.toUpperCase() && /[A-Za-z]/.test(message)) {
    // If the message ends with a question mark, it's considered shouting in a question
    if (message.endsWith('?')) {
      return 'Calm down, I know what I\'m doing!';
    }
    return 'Whoa, chill out!';
  }

  // If the message is a question (ends with a question mark)
  if (message.trim().endsWith('?')) {
    return 'Sure.';
  }

  // For any other case, return "Whatever."
  return 'Whatever.';
};

  

File: solutions/java/bob/1/src/main/java/Bob.java
class Bob {

        public String hey(String input) {
        if (input == null || input.trim().isEmpty()) {
            return "Fine. Be that way!";
        }

        input = input.trim();
        boolean isQuestion = input.endsWith("?");
        boolean hasLetters = input.chars().anyMatch(Character::isLetter);
        boolean isYelling = hasLetters && input.equals(input.toUpperCase());

        if (isYelling && isQuestion) {
            return "Calm down, I know what I'm doing!";
        } else if (isYelling) {
            return "Whoa, chill out!";
        } else if (isQuestion) {
            return "Sure.";
        } else {
            return "Whatever.";
        }
    }

}